### ISSUE TITLE
[Feature] Ignore a line/rule

### ISSUE THREAD

**Issue by adamzimmermann** [VincentLanglet/Twig-CS-Fixer#5]
Can you add documentation to the README that explains how one can ignore a line(s)s of code or disable a ruleset all together during a check? I'm happy to contribute that if you can point me to some examples or even provide them here.

### Expected behavior

Ability ignore certain lines/rules.

### Actual behavior

I'm unable to do so, or haven't found the documentation.

**Comment by adamzimmermann** [comment]
I realized I wasn't sure how to do this, when I encountered https://github.com/VincentLanglet/Twig-CS-Fixer/issues/6 and wanted to ignore it until we had a permanent fix in place.

**Comment by VincentLanglet** [comment]
I would say that unfortunately, currently there is no way to ignore a line/rule 😅
I was fixing bug as soon as I got them in my project.

**Comment by adamzimmermann** [comment]
Thanks for your quick response.

---

> I was fixing bug as soon as I got them in my project.

Haha, I like that approach!

---

I'll put this on my todo list and see if I can find some time to contribute something if we continue using this.

**Comment by VincentLanglet** [comment]
> I'll put this on my todo list and see if I can find some time to contribute something if we continue using this.

I'll be happy to discuss about it if you continue using this.

Currently
- The rules used are defined here: https://github.com/VincentLanglet/Twig-CS-Fixer/blob/main/src/Command/TwigCsFixerCommand.php#L71-L72
- The files used are determined here https://github.com/VincentLanglet/Twig-CS-Fixer/blob/main/src/Command/TwigCsFixerCommand.php#L74 (I don't remember why I didn't used the Symfony finder).
- The report is generated here: https://github.com/VincentLanglet/Twig-CS-Fixer/blob/main/src/Command/TwigCsFixerCommand.php#L79, I could pass a list of message to ignore.

A config object with
- The ruleset to use
- The files to use
- The errors to ignore
returned by a `.twig_cs.php` file could be a solution.

**Comment by VincentLanglet** [comment]
Solved, see https://github.com/VincentLanglet/Twig-CS-Fixer#custom-configuration

**Comment by adamzimmermann** [comment]
🎉 thank you!

**Comment by VincentLanglet** [comment]
> 🎉 thank you!

You welcome, don't hesitate if you have other idea of feature.
And I can help if you want to contribute.

**Comment by AlexSkrypnyk** [comment]
@VincentLanglet 
The custom configuration section in the readme does not describe how to ignore a single line **inline**.

Is it possible to add this functionality? I'm happy to open a new issue and sponsor this work.

**Comment by VincentLanglet** [comment]
Currently this is not possible but I would love having this feature so feel free opening an issue/sponsoring it.

I'll have busy weeks this month but I'll try to find time (or maybe someone else will !)

**Comment by AlexSkrypnyk** [comment]
@VincentLanglet 
Sponsored :) happy to do again to get this working

**Comment by VincentLanglet** [comment]
> @VincentLanglet Sponsored :) happy to do again to get this working

Thanks a lot @AlexSkrypnyk.

The feature is working on the branch https://github.com/VincentLanglet/Twig-CS-Fixer/pull/162
```twig
{# twig-cs-fixer-disable SomeRule #} This disable the rule on the whole file
{# twig-cs-fixer-disable-line SomeRule #} This disable the rule on the line of the `{#`
{# twig-cs-fixer-disable-next-line SomeRule #} This disable the rule on the line after the `#}`
```
There you can also be more specific with the syntax `Rule.ErrorIdentifier.TokenName:LineNumber:LinePosition`
```twig
{# twig-cs-fixer-disable SomeRule.SomeErrorIdentifier:42 #} This disable a specific error of the rule on the line 42.
{# twig-cs-fixer-disable SomeRule.SomeErrorIdentifier.TokenName #} This disable a specific error of the rule for the whole file but only for the token "TokenName".
{# twig-cs-fixer-disable-line SomeRule.SomeErrorIdentifier::2 #} This disable a specific error of the rule on the current line but only for the second token of the line.
{# twig-cs-fixer-disable SomeRule.SomeErrorIdentifier.TokenName #} This disable a specific error of the rule on the current line but only for the second token of the line.
```
and so on...

What I mainly miss, now is to write documentation.

**Comment by AlexSkrypnyk** [comment]
@VincentLanglet 
Awesome!
Can the `twig-cs-fixer-disable` be used without specifying any rules? this would be really good to have as well.

The documentation above is sufficient, imo.

**Comment by VincentLanglet** [comment]
I didnt think about this usage, but this should be possible to add it I think. I'll try to find time the next week.

**Comment by VincentLanglet** [comment]
@AlexSkrypnyk The work on the https://github.com/VincentLanglet/Twig-CS-Fixer/pull/162/files#diff-79d8f8cb20a165e13d56d6eac97179fc2db0a457c1bf06128e05474efa20a121 PR seems finished to me.

I would be happy to have a look from you:
- If you can test the branch to know if it works as you expect it
- To know if the doc from https://github.com/VincentLanglet/Twig-CS-Fixer/pull/162/commits/93cd8ef32224e5f019733200cb7a841dd8f7c635 is clear

**Comment by AlexSkrypnyk** [comment]
@VincentLanglet 
The doc looks great!

I do not have the capacity to test this now, sorry. Please do not wait on me.

Can you please connect with me on X or LinkedIn so I can send you a PM re sponsoring.
